aiverify-foundation/moonshot
# Ms 817 manage config api

## Description

Currently, the only way for users to edit the configuration of attack modules and metrics is to edit the configurations directly. This PR is a new API for users to manage these configurations. Users can edit and delete the configurations of attack modules and metrics.

## Motivation and Context

Create API for users to easily manage the configurations. However, this is only available on the API and web API layer. There is no interface on the UI and CLI to access this yet.

## Type of Change

A new feature

## How to Test

1. Use Postman or cURL to access the API endpoints. The endpoints:
    - http://127.0.0.1:5000/api/v1/attack-modules/update-config
    - http://127.0.0.1:5000/api/v1/attack-modules/delete-config
    - http://127.0.0.1:5000/api/v1/metric/update-config
    - http://127.0.0.1:5000/api/v1/metric/delete-config

2. To test both the `update-config` endpoints: 
    - Send a POST request with the parameters attack_module_id=your_target_am_id and metric_id=your_target_me_id to the attack module and metric endpoints respectively 
    -  In the request, send raw data in JSON format to update the configuration. Example: 
     ```
    {
        "endpoints": ["my-updated-endpoint"]
    }
    ```
    - You should see that the configuration has been updated in the configuration file

3. To test both the `delete-config` endpoints: 
    - Send a POST request with the parameters attack_module_id=something and metric_id=something to the attack module and metric endpoints respectively 
    - You should see that the configuration has been deleted in the configuration file

## Checklist

Please check all the boxes that apply to this pull request using "x":


## Screenshots (if applicable)

x

## Additional Notes

x

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
